glean-core/src/upload/mod.rs

```diff
@@ -94,99 +123,172 @@ impl PingUploadManager {
         queue.clear();
     }
 
-    /// Get the next `PingRequest` in queue.
-    pub fn get_next_ping(&self) -> Option<PingRequest> {
+    /// Gets the next `PingRequest` in queue.
+    ///
+    /// # Return value
+    ///
+    /// `PingUploadTask` - see [`PingUploadTask`](enum.PingUploadTask.html) for more information.
+    pub fn get_next_ping(&self) -> PingUploadTask {
+        if !self.has_processed_pings_dir() {
+            return PingUploadTask::Wait;
+        }
+
         let mut queue = self
             .queue
             .write()
             .expect("Can't write to pending pings queue.");
-        queue.pop_front()
+        match queue.pop_front() {
+            Some(request) => PingUploadTask::Upload(request),
+            None => PingUploadTask::Done,
+        }
     }
 }
 
 #[cfg(test)]
 mod test {
+    use std::time::Duration;
+    use std::thread;
+
+    use serde_json::json;
+
+    use super::*;
     use crate::metrics::PingType;
-    use serde_json::Value as JsonValue;
+    use crate::tests::new_glean;
 
-    let uuid = "c0ffeec0-ffee-c0ff-eec0-ffeec0ffeec0";
-    let url = "http://example.com";
-    let body = json!("{}")
+    const UUID: &str = "c0ffeec0-ffee-c0ff-eec0-ffeec0ffeec0";
```

Comment:
nit: thinking of this a bit, can we change its value? This has other meanings within the glean repo, so let's use a different fake UUID.

Reply:
Didn't you change that? Was that change somehow lost again?

Reply:
Hm, I made some mistake on rebasing this... Thanks for noticing.